sonar-xml-plugin/src/main/java/org/sonar/plugins/xml/XmlSensor.java

```diff
@@ -171,13 +167,13 @@ private static void processParseException(ParseException e, SensorContext contex
     }
   }
 
-  private static void processException(RuntimeException e, SensorContext context, InputFile inputFile) {
+  private static void processException(Exception e, SensorContext context, InputFile inputFile) {
     reportAnalysisError(e, context, inputFile);
 
     throw new IllegalStateException("Unable to analyse file " + inputFile.uri(), e);
```

Comment:
Ok as you need to rework the PR, i would go for a log instead of throwing the ISE here.